api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -46,10 +57,70 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 				case FhirConstants.COMMON_SEARCH_HANDLER:
 					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
 					break;
+				case FhirConstants.HAS_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleHasAndListParam(criteria, (HasAndListParam) param.getParam()));
+					break;
```

Comment:
I expect the way FHIR handles the _has parameters is that there will always be only 0 or 1 of these (multiple params are combined into the "andList" portion of this).